### PR TITLE
tests: benchmarks: Verify power management for UARTE  (using s2ram)

### PR DESCRIPTION
**Pull request by nordic-bami**
Verify PM ability to handle UARTE  power management This sample uses GPIO loopback

### CODE REVIEW

File: tests/benchmarks/multicore/idle_uarte/prj.conf [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+CONFIG_PM_S2RAM=y
+CONFIG_PM_S2RAM_CUSTOM_MARKING=y
+CONFIG_PM_DEVICE=y
+CONFIG_PM_DEVICE_RUNTIME=y
+CONFIG_POWEROFF=y
+
+CONFIG_GPIO=n
+CONFIG_BOOT_BANNER=n
+CONFIG_NRFS_MRAM_SERVICE_ENABLED=n
+CONFIG_CLOCK_CONTROL=n
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_uarte/prj.conf:16]: not needed

File: tests/benchmarks/multicore/idle_uarte/boards/nrf54h20dk_nrf54h20_common.dtsi [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+&cpuapp_dma_region {
+	status="okay";
+};
+
+&pinctrl {
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_uarte/boards/nrf54h20dk_nrf54h20_common.dtsi:11]: Please add comment with info what has changed from original configuration or why.

File: tests/benchmarks/multicore/idle_uarte/prj.conf [hunk 3]
```diff
@@ -0,0 +1,22 @@
+CONFIG_SERIAL=y
+CONFIG_MAIN_STACK_SIZE=2048
+CONFIG_UART_USE_RUNTIME_CONFIGURE=y
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_uarte/prj.conf:3]: The default is y, why adding it?

File: tests/benchmarks/multicore/idle_uarte/prj.conf [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+CONFIG_UART_ASYNC_API=y
+
+CONFIG_PM=y
+CONFIG_PM_S2RAM=y
+CONFIG_PM_S2RAM_CUSTOM_MARKING=y
+CONFIG_PM_DEVICE=y
+CONFIG_PM_DEVICE_RUNTIME=y
+CONFIG_POWEROFF=y
+
+CONFIG_GPIO=n
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_uarte/prj.conf:13]: This is the default, not needed?

File: tests/benchmarks/multicore/idle_uarte/prj.conf [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+CONFIG_BOOT_BANNER=n
+CONFIG_NRFS_MRAM_SERVICE_ENABLED=n
+CONFIG_CLOCK_CONTROL=n
+
+# Enable for debugging purposes only
+CONFIG_PRINTK=n
+CONFIG_LOG=n
+CONFIG_CONSOLE=n
+CONFIG_UART_CONSOLE=n
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_uarte/prj.conf:22]: not needed

File: tests/benchmarks/multicore/idle_uarte/src/main.c [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+	err = uart_callback_set(uart_dev, async_uart_callback, NULL);
+	if (err != 0) {
+		printk("Unexpected error when setting callback %d\n", err);
+		return -1;
+	}
+
+	while (1) {
+		printk("Hello\n");
+
+		uart_rx_enable(uart_dev, test_buffer, TEST_BUFFER_LEN,
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_uarte/src/main.c:101]: Please check error codes from those function and return if error occurs

File: tests/benchmarks/multicore/idle_uarte/prj.conf (hunk 5 above)
Comment (nordic-bami) [reply tests/benchmarks/multicore/idle_uarte/prj.conf:22]: This is left here for convenience only, if debugging is desired.

**Review by nordic-piks — APPROVED** [review]

**Review by nordic-pikr — APPROVED** [review]

File: tests/benchmarks/multicore/idle_uarte/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,18 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+if (NOT SYSBUILD)
```
Comment (nordicjm) [inline tests/benchmarks/multicore/idle_uarte/CMakeLists.txt:10]: as per other PR

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Review by nordicjm — APPROVED** [review]

**Pull request merged**